R/git.R

```diff
@@ -135,10 +135,10 @@ github_remote_parse <- function(x) {
 
   if (grepl("^https", x)) {
     # https://github.com/hadley/devtools.git
-    re <- "github.com/(.*?)/(.*)\\.git"
+    re <- "github[^/]*/(.*?)/(.*)\\.git"
   } else if (grepl("^git", x)) {
     # git@github.com:hadley/devtools.git
-    re <- "github.com:(.*?)/(.*)\\.git"
+    re <- "github[^:]*:(.*?)/(.*)\\.git"
```

Comment:
Here's what I have for L138 and L141:

```R
    re <- "github[^/]*/(.*?)/(.*)\\.git" # L138
    re <- "github[^:]*:(.*?)/(.*)\\.git" # L141
```

I think the difference is in the delimiter, `/` for L138 and `:` for L141, to distinguish

```
https://github.com/hadley/devtools.git # L138 case
git@github.com:hadley/devtools.git     # L141 case
```

I think the two might be combined, as I don't think that either delimiter is legal part of a hostname in either case.

What do you think?